### ISSUE TITLE
Documentation not clear on torch.expand() alternatives when performing torch.autograd.gradcheck

### ISSUE THREAD

**Issue by nwschurink** [pytorch/pytorch#12285]
I am writing a custom loss funtion in which I expanded an array to several duplicates using unsqueeze() and the torch.expand() function. e.g. Tensor(1,1,100,100,100)-> Tensor(1,1,100,100,100,12).

My torch.autograd.gradcheck is failing (without pointing out why exactly), however looking in the documentation I see that it might be related to this tensor expansion. However, the documentation does not give any additional suggestions about how to resolve around this issue..

To obtain a similar tensor expansion without gradcheck to fail, should I then use torch.cat and call it several times instead?? Memory wise this doesn't sound like the smartest move as I am then actually copying the same data several times in memory right?. 

Would you suggest to use the cat funtion during the gradcheck and then change it to an expansion when used for training? Or is the expansion not at all possible within a loss function?

Hope to hear your toughts.

**Comment by SsnL** [comment]
The note only talks about input though. It should have nothing to do with how your function is implemented. All you need to make sure is that the input used in gradcheck don't have overlapping storage.

**Comment by nwschurink** [comment]
Ah you are right, I misunderstood the reference. Sorry for the misunderstanding ;)
